woocommerce/woocommerce-ios
# Login enhancement: Extract plugin setup web view model from view controller

Part of #7236 


### Description

For #7236, we want to add a new screen for the error when a user tries to log in to a site that doesn't have WooCommerce. For this screen, we want to add a button to install Woo on a web view. This will be very similar to the flow with installing Jetpack, so this PR aims to extract the view model for Jetpack setup from the web view controller so that we can reuse the view controller logic later with Woo setup.

This PR also creates an extension of `WKWebView` and extracts some logic from `AuthenticatedWebView` to authenticate WPCom automatically and avoid forcing users to log in twice. I haven't removed the logic from `AuthenticatedWebView` though since that will require regression test for shipping labels, so I'll create a separate task for the migration after this PR is merged.

### Testing instructions

We need to make sure that the Jetpack setup flow still works as before:
- Prerequisite: make sure to have a test site with WooCommerce and no Jetpack.
- To make sure any cookies are cleared from the app's browser, delete the app first.
- Select Enter site address.
- Proceed to log in with either WP.com credentials. After the login succeeds, notice that you reach the no Jetpack error screen.
- Select Install Jetpack and follow the instruction on the web view. 
- When you reach the Jetpack Authorization step, notice that you no longer have to enter WPCom credentials in the web view.
- After the authorization step, notice that the web view is dismissed, and the app navigates to the home screen.

### Screenshot

https://user-images.githubusercontent.com/5533851/182359228-84f5fbc6-62a2-4923-83af-799f648ee54f.mp4



---
